src/cipher/dune

```diff
@@ -3,6 +3,7 @@
  (name dream__cipher)
  (libraries
   cstruct
+  logs
```

Comment:
Can you simplify the pattern? We can essentially guarantee during "composition" that the `ref` will be initialized. I don't think it's worth picking up a dependency on `logs` in `dream.cipher` for the sake of somewhat elaborate messages about initialization problems :) I suggest deleting all natural-language messages and just doing a simple `ref`, dereferencing of it, and setting of it in `dream.ml`, only.